homeassistant/components/opnsense/device_tracker.py

```diff
@@ -0,0 +1,67 @@
+"""Device tracker support for OPNSense routers."""
+import logging
+
+from homeassistant.components.device_tracker import DeviceScanner
+
+from homeassistant.components.opnsense import OPNSENSE_DATA
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_get_scanner(hass, config, tracker_interfaces=None):
+    """Configure the OPNSense device_tracker."""
+    interface_client = hass.data[OPNSENSE_DATA]["interfaces"]
+    interfaces = tracker_interfaces or ["LAN"]
+    scanner = OPNSenseDeviceScanner(interface_client, interfaces)
+    return scanner
+
+
+class OPNSenseDeviceScanner(DeviceScanner):
+    """This class queries a router running ASUSWRT firmware."""
+
+    # Eighth attribute needed for mode (AP mode vs router mode)
+    def __init__(self, client, interfaces):
+        """Initialize the scanner."""
+        self.last_results = {}
+        self.success_init = False
+        self.client = client
+        self.interfaces = interfaces
+
+    def _get_mac_addrs(self, devices):
```

Comment:
Docstring is missing.

Reply:
Added in: https://github.com/home-assistant/home-assistant/pull/26834/commits/914fce5e613221f69c8fcb7edbcf20a97b0c418d